object-manipulation-2/src/pick.js

```diff
@@ -0,0 +1,34 @@
+/* exported pick */
+
+function pick(source, keys) {
+
+  var output = {};
+
+  for (var key in source) {
+
+    if (source[key] !== undefined) {
+      for (var i = 0; i < keys.length; i++) {
```

Comment:
I would swap the order of your loops here. Instead of doing a full loop of the `keys` array for each property in `source`, I would loop over the `keys` array and check if the current `key` is a property with value in `source`. It will end up requiring far less operations.